balances/src/lib.rs

```diff
@@ -111,6 +111,19 @@ pub mod pallet {
 			Self::deposit_event(Event::FeeConversionFactorUpdated(fee_conversion_factor));
 			Ok(().into())
 		}
+
+		#[pallet::weight((<T as Config>::WeightInfo::transfer_all(), DispatchClass::Normal))]
+		pub fn transfer_all(
+			origin: OriginFor<T>,
+			dest: T::AccountId,
+			cid: CommunityIdentifier,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+			let amount = Self::balance(cid, &from);
+			Self::do_transfer(cid, from.clone(), dest, amount)?;
+			Self::remove_account(cid, &from)?;
```

Comment:
This should be incorporated into `do_transfer` we also want to reap the account if the balance falls below the ED in normal transfers.

Reply:
should it be in `do_transfer` or maybe rather in `balance_entry_updated` to make sure that this mechanism also kicks in when demurrage puts an account below ED? and generally, should this be part of this PR?

Reply:
I think in `do_transfer` suffices for now because otherwise the demurrage is only applied when you get money. So the reaping is most likely not kicking in anyhow, and it might be a bit more tricky there to not make any errors.

Reply:
ok done